Assets/_Game/Scripts/Checkpoint/CheckpointTrigger.cs

```diff
@@ -21,14 +22,19 @@ public class CheckpointTrigger : MonoBehaviour
     private void Awake()
     {
         candleController = GameObject.FindGameObjectWithTag("Candle").GetComponent<CandleController>();
+        activatePopup = transform.GetChild(0).gameObject;
         anim = GetComponent<Animator>();
-        light = GetComponent<Light2D>();
+        light2D = GetComponent<Light2D>();
```

Comment:
Same here, if you're serializing, just assign it in the inspector rather than overwriting whatever is there on awake

Reply:
Removed

Reply:
You still have the light and the animator, both of which can be assigned and don't need a GetComponent call